contracts/gauge/child_gauge/factory/factory_v_201.vy

```diff
@@ -0,0 +1,364 @@
+    """
+    self._psuedo_mint(_gauge, msg.sender)
+
+
+@external
+@nonreentrant("lock")
+def mint_many(_gauges: address[32]):
+    """
+    @notice Mint everything which belongs to `msg.sender` across multiple gauges
+    @param _gauges List of `LiquidityGauge` addresses
+    """
+    for i in range(32):
+        if _gauges[i] == empty(address):
+            pass
+        self._psuedo_mint(_gauges[i], msg.sender)
+
+
+@external
+def deploy_gauge(_lp_token: address, _salt: bytes32, _manager: address = msg.sender) -> address:
+    """
+    @notice Deploy a liquidity gauge
+    @param _lp_token The token to deposit in the gauge
+    @param _salt A value to deterministically deploy a gauge
+    @param _manager The address to set as manager of the gauge
+    """
+    if self.get_gauge_from_lp_token[_lp_token] != empty(address):
+        # overwriting lp_token -> gauge mapping requires
+        assert msg.sender == self.owner  # dev: only owner
+
+    gauge_data: uint256 = 1  # set is_valid_gauge = True
+    implementation: address = self.get_implementation
+    salt: bytes32 = keccak256(_abi_encode(chain.id, _salt))
+    gauge: address = create_minimal_proxy_to(
+        implementation, salt=salt
+    )
+
+    if msg.sender == self.call_proxy:
+        gauge_data += 2  # set mirrored = True
+        log UpdateMirrored(gauge, True)
+        # issue a call to the root chain to deploy a root gauge
+        CallProxy(self.call_proxy).anyCall(
+            self,
+            _abi_encode(chain.id, _salt, method_id=method_id("deploy_gauge(uint256,bytes32)")),
+            empty(address),
+            1
+        )
+
+    self.gauge_data[gauge] = gauge_data
+
+    idx: uint256 = self.get_gauge_count
+    self.get_gauge[idx] = gauge
+    self.get_gauge_count = idx + 1
+    self.get_gauge_from_lp_token[_lp_token] = gauge
+
+    # derive root gauge address
+    gauge_codehash: bytes32 = keccak256(
+        concat(
+            0x602d3d8160093d39f3363d3d373d3d3d363d73,
+            convert(self.root_implementation, bytes20),
+            0x5af43d82803e903d91602b57fd5bf3,
+        )
+    )
+    digest: bytes32 = keccak256(concat(0xFF, convert(self.root_factory, bytes20), salt, gauge_codehash))
+    root: address = convert(convert(digest, uint256) & convert(max_value(uint160), uint256), address)
+
+    # If root is uninitialized, self.owner can always set the root gauge manually
+    # on the gauge contract itself via set_root_gauge method
+    ChildGauge(gauge).initialize(_lp_token, root, _manager)
+
+    log DeployedGauge(implementation, _lp_token, msg.sender, _salt, gauge)
+    return gauge
+
+
+@external
+def set_crv(_crv: ERC20):
+    """
+    @notice Sets CRV token address
+    @dev Child gauges reference the factory to fetch CRV address
+         If empty, the gauges do not mint any CRV tokens.
+    @param _crv address of CRV token on child chain
+    """
+    assert msg.sender == self.owner
```

Comment:
Add additional check if self.crv is empty or not and only proceed if empty

Reply:
well we only want to set crv once because gauges reference factory for crv address. so if a gauge already uses a crv address for emissions, we shouldnt change that address somehow.